app/services/orangelight_document/coin_builder.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+class OrangelightDocument
+  class CoinBuilder
+    attr_reader :decorator, :parent
+    def initialize(decorator)
+      @decorator = decorator
+      @parent = decorator.decorated_parent
+    end
+
+    def build
+      clean_document(document_hash)
+    end
+
+    private
+
+      def clean_document(hash)
```

Comment:
I was thinking about whether this class should go in `app/resources/coins`, and one of the benefits of having them all in `app/services/orangelight_document` is that it would make it easier to build common functionality in one place.  Do you want to start doing that, e.g., by putting `build` and `clean_document` in `OrangelightDocument` and making this a subclass?

Reply:
I don't think CoinBuilder is a subclass of OrangelightDocument. However, I could see adding a base builder class that has utility methods like `clean_document`. I'm on the fence about moving it into `app/resources/coins/`. Since we would be removing it from it's namespace, we could rename it to `CoinOrangelightBuilder` or `OrangelightCoinBuilder` to better indicate its function.

Reply:
@escowles On further reflection, putting builders in the `app/resources/*` would be consistent with our current pattern. What do you think?

Reply:
I do like the idea of moving them into `app/resources/*` — and we can probably wait to create the base class until we have another type with an Orangelight builder.

Reply:
👍 I'll re-arrange the classes and push up an update.

Reply:
Updated.